pylint/checkers/classes/class_checker.py

```diff
@@ -904,6 +911,33 @@ def _check_proper_bases(self, node: nodes.ClassDef) -> None:
                         confidence=INFERENCE,
                     )
 
+            if isinstance(ancestor, nodes.ClassDef) and ancestor.is_subtype_of(
```

Comment:
`visit_classdef` is quite bloated already, would you mind moving the new code to a `check_flag` (better name maybe) method?

Reply:
Would it be OK to relocate the `enum.Enum` code at the same time (`_check_enum_bases`, or similar)?  Reasoning: it might be possible to de-duplicate some of the conditional logic after doing that.

Reply:
yeah for sure!

Reply:
I'm experimenting with this a bit locally; it's making me wonder whether it'd be possible (and more efficient) to use the existing tree `visit_` handlers to achieve the same check.

Reply:
(maybe: for now, it seems simpler and likely more maintainable to keep the logic grouped textually)